Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Service_Ownership.yml

```diff
@@ -464,58 +242,129 @@ tasks:
       type: condition
       iscommand: false
       brand: ""
     nexttasks:
       '#default#':
+      - "10"
+      GCP:
       - "6"
+    separatecontext: false
+    conditions:
+    - label: GCP
+      condition:
+      - - operator: containsGeneral
+          left:
+            value:
+              complex:
+                root: ASM.ExternalService
+                accessor: externally_detected_providers
+            iscontext: true
+          right:
+            value:
+              simple: Google
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": -50,
+          "y": 590
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "15":
+    id: "15"
+    taskid: 25035e9c-31cb-46bc-86b3-9faa72de87a8
+    type: condition
+    task:
+      id: 25035e9c-31cb-46bc-86b3-9faa72de87a8
+      version: -1
+      name: Was there a result?
+      description: Determines if there was a result from the previous command to continue cloud enrichment.
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#default#':
+      - "10"
       "yes":
-      - "2"
+      - "14"
     separatecontext: false
     conditions:
     - label: "yes"
       condition:
-      - - operator: isExists
+      - - operator: isNotEmpty
           left:
             value:
-              complex:
-                root: modules
-                filters:
-                - - operator: isEqualString
-                    left:
-                      value:
-                        simple: modules.brand
-                      iscontext: true
-                    right:
-                      value:
-                        simple: GCP-IAM
-                - - operator: isEqualString
-                    left:
-                      value:
-                        simple: modules.state
-                      iscontext: true
-                    right:
-                      value:
-                        simple: active
+              simple: ASM.ExternalService
             iscontext: true
           right:
             value: {}
     continueonerrortype: ""
     view: |-
       {
         "position": {
-          "x": 460,
-          "y": 380
+          "x": -50,
+          "y": 340
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "16":
+    id: "16"
+    taskid: 05ce1338-5084-49af-8745-a83af6781f93
+    type: regular
+    task:
+      id: 05ce1338-5084-49af-8745-a83af6781f93
+      version: -1
+      name: Get external service information
+      description: Get service details according to the service ID.
+      script: Cortex Attack Surface Management|||asm-get-external-service
```

Comment:
This command is already run in "Cortex ASM - Enrichment".  So we can do one of the following:

- Pass full results of ${ASM.ExternalService.externally_detected_providers} into this playbook as an input (preferred)
- Add a conditional to make sure CASM integration is enabled (similar to "Is Cortex ASM enabled?" task in "Cortex ASM - Enrichment".

Reply:
I was thinking about this repetition as well. The first option should be a later change, just to keep the main playbook out of scope. I'll go with the second option.

Reply:
[6e27a2f](https://github.com/demisto/content/pull/26334/commits/6e27a2fbc533642e2118f899ab3b10bf3a760ac6)